packages/syft/src/syft/service/worker/worker_image_service.py

```diff
@@ -0,0 +1,151 @@
+
+    def __init__(self, store: DocumentStore) -> None:
+        self.store = store
+        self.stash = SyftWorkerImageStash(store=store)
+
+    @service_method(
+        path="worker_image.submit_dockerfile",
+        name="submit_dockerfile",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def submit_dockerfile(
+        self, context: AuthedServiceContext, docker_config: DockerWorkerConfig
+    ) -> Union[SyftSuccess, SyftError]:
+        worker_image = SyftWorkerImage(
+            config=docker_config,
+            created_by=context.credentials,
+        )
+        res = self.stash.set(context.credentials, worker_image)
+
+        if res.is_err():
+            return SyftError(message=res.err())
+
+        return SyftSuccess(
+            message=f"Dockerfile <id: {worker_image.id}> successfully submitted."
+        )
+
+    @service_method(
+        path="worker_image.build",
+        name="build",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def build(
+        self,
+        context: AuthedServiceContext,
+        uid: UID,
+        tag: str,
+        push: bool = False,
+        container_registry: Optional[str] = None,
+    ) -> Union[SyftSuccess, SyftError]:
+        result = self.stash.get_by_uid(credentials=context.credentials, uid=uid)
+        if result.is_err():
+            return SyftError(
+                message=f"Failed to get image for uid: {uid}. Error: {result.err()}"
+            )
+
+        worker_image: SyftWorkerImage = result.ok()
+
+        try:
+            image_tag = SyftWorkerImageTag.from_str(full_str=tag)
+        except pydantic.ValidationError as e:
+            return SyftError(message=f"Failed to create tag: {e}")
+
+        worker_image.image_tag = image_tag
+        worker_image, result = build_using_docker(worker_image=worker_image, push=push)
+
+        if isinstance(result, SyftError):
+            return result
+
+        update_result = self.stash.update(context.credentials, obj=worker_image)
+
+        if update_result.is_err():
+            return SyftError(
+                message=f"Failed to update image meta information: {update_result.err()}"
+            )
+
+        return result
+
+    @service_method(
+        path="worker_image.list",
+        name="list",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def list(
+        self, context: AuthedServiceContext
+    ) -> Union[List[SyftWorkerImage], SyftError]:
+        """
+        One image one docker file for now
+        TODO: change repr when listing
+        """
+        result = self.stash.get_all(credentials=context.credentials)
+        if result.is_err():
+            return SyftError(message=f"{result.err()}")
+
+        return result.ok()
+
+    @service_method(
+        path="worker_image.delete",
+        name="delete",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def delete(
+        self, context: AuthedServiceContext, uid: UID, clear_image: bool = False
+    ) -> Union[SyftSuccess, SyftError]:
```

Comment:
I think we should always remove the image with `delete`, getting rid of the `clear_image` arg altogether. When an user do `delete` that's probably what they mean. I don't see a use case for deleting the image entry from the stash but leaving the image there. Deleting the image entry but not the image itself would lead to further complications later on (`e.g.` dangling images).

Reply:
I agree @kiendang. Fixed the code accordingly